met-web/src/components/engagement/form/EngagementWidgets/WhoIsListening/WhoIsListeningContext.tsx

```diff
@@ -41,13 +58,44 @@ export const WhoIsListeningProvider = ({ children }: { children: JSX.Element | J
     const { savedEngagement } = useContext(ActionContext);
     const [getContactsTrigger] = useLazyGetContactsQuery();
     const dispatch = useAppDispatch();
+    const { widgets } = useContext(WidgetDrawerContext);
+    const widget = widgets.find((widget) => widget.widget_type_id === WidgetType.WhoIsListening) || null;
 
     const [contactToEdit, setContactToEdit] = useState<Contact | null>(null);
     const [addContactDrawerOpen, setAddContactDrawerOpen] = useState(false);
     const [contacts, setContacts] = useState<Contact[]>([]);
+    const [listeningWidget, setListeningWidget] = useState<ListeningWidget>(emptyListeningWidget);
     const [addedContacts, setAddedContacts] = useState<Contact[]>([]);
     const [loadingContacts, setLoadingContacts] = useState(true);
 
+    if (!widget) {
```

Comment:
Just curious about this part - why did we need to check if there was no widget and then return a different context provider?

Reply:
The widget is passed down as a prop, and if it's missing, it just displays a blank widget. That's the way it was designed before. We could change it if needed. I have moved the useEffect to the top of the component to avoid the errors.

Reply:
Just to confirm what we discussed over Teams, it's a useContext, not a prop, that it's passed down from. I know that I didn't create this code, but I couldn't find it on main, so I removed it, made some additional conditions, and tested again. I'm having no errors or issues now, so this code has been removed.